### PR TITLE
Releases in absentia

### PR DESCRIPTION
**Pull request by nabijaczleweli**
[`v1.12.6`](https://github.com/thecoshman/http/releases/tag/v1.12.6)
[`v1.13.0`](https://github.com/thecoshman/http/releases/tag/v1.13.0)
[`v1.13.1`](https://github.com/thecoshman/http/releases/tag/v1.13.1)
[`v1.13.2`](https://github.com/thecoshman/http/releases/tag/v1.13.2) (tentative)

### CODE REVIEW

**Review by thecoshman — CHANGES_REQUESTED** [review]

File: appveyor.yml [hunk 1]
```diff
@@ -9,20 +12,8 @@ clone_folder: C:\http
...
-  - pacman --noconfirm -U C:\zstd.txz
-  - pacman --noconfirm -U C:\pacman.txz
-  - pacman --noconfirm -U C:\msys2-keyring.tzst
-  - bash -lc "pacman --needed --noconfirm -Sy pacman-mirrors"
-  - bash -lc "pacman --noconfirm -Sy pacman"
-  - bash -lc "pacman --noconfirm -Sy"
-  - bash -lc "pacman --noconfirm -Su"
-  - bash -lc "pacman --noconfirm -Sy"
+  - bash -lc "pacman --noconfirm -Syyu"
+  - bash -lc "pacman --noconfirm -Syyu"
```
[the remark this answers is not in the document]
Comment (nabijaczleweli) [reply appveyor.yml:16]: Double upgrade required for some reason, yes.

File: http.md [hunk 2]
```diff
@@ -126,6 +126,15 @@ pass parameters like what port to use.
 
     Can be specified any amount of times. Default: none.
 
+  --proxy-redir [HEADER-NAME:CIDR]
```
[the remark this answers is not in the document]
Comment (nabijaczleweli) [reply http.md:129]: It's for when you have a
```
location / {
    proxy_pass http://127.0.0.1:8211/some-different-URI-root$request_uri;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Original-URL https://$server_name$request_uri;
}
```

We already handled X-Forwarded-For, but it's just for logging:
```
[2023-08-26 12:07:10] 127.0.0.1:49258 for 95.181.232.30, 188.114.111.21 was served file /mnt/filling/store/nabijaczleweli/debian.nabijaczleweli.xyz/dists/bookworm/InRelease as text/plain
```

In this example let's say we have `http /mnt/filling/store/nabijaczleweli/debian.nabijaczleweli.xyz/dists` and
```
location ~ /dists(.*) {
    proxy_pass http://127.0.0.1:8000$1;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Original-URL https://$server_name$request_uri;
}
```

If, in this example, someone requests `https://debian.nabijaczleweli.xyz/dists/bookworm` we need to redirect them to `https://debian.nabijaczleweli.xyz/dists/bookworm/` and return the generated index for that URL.

Without this flag, we redirected to `/bookworm/` (which is equivalent to `https://debian.nabijaczleweli.xyz/bookworm`, and wrong).

File: src/ops/mod.rs [hunk 3]
```diff
@@ -742,8 +744,27 @@ impl HttpHandler {
...
+    fn slashise(u: String) -> String {
+        let mut b = u.into_bytes();
+        b.insert(b.iter().position(|&c| c == b'?').unwrap_or(b.len()), b'/');
+        unsafe { String::from_utf8_unchecked(b) }
+    }
+
     fn handle_get_dir_index_no_slash(&self, req: &mut Request, idx_ext: &str) -> IronResult<Response> {
-        let new_url = req.url.to_string() + "/";
+        let mut new_url = None;
+        for (network, header) in &self.proxy_redirs {
```
[the remark this answers is not in the document]
Comment (nabijaczleweli) [reply src/ops/mod.rs:755]: Lifted

File: appveyor.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (thecoshman) [reply appveyor.yml:16]: Ergh... a stupid thing but I guess not much we can do about it. Maybe a comment to explain that it actually is needed

**Pull request merged**